docs/testsuite_overview.md

```diff
@@ -0,0 +1,269 @@
+* **responses**: input HTTP responses for wire protocol tests (Rest.li protocol 1)
+* **responses-v2**: input HTTP responses for wire protocol tests (Rest.li protocol 2)
+* **restspecs**: restspecs generated from Java resources. These should be used by Rest.li implementations
+to make request builders.
+* **snapshots**: snapshots for resources
+
+#### Java TestNG Tests
+In ```src/test/java```, you can find the Java test suite, which uses the TestNG testing framework. 
+You can add test suites for new languages in ```src/test/```. The Java suite is separated into two folders:
+* **test**: Java files containing the tests, and utility methods for running tests. 
+```TestRestClientAgainstStandardTestSuite``` and ```TestRestClientWithManualAssertions``` contain the tests,
+while ```StandardTestSuiteBase``` has utility methods that load files and compare requests.
+
+* **testsuite**: Java files for building and loading requests and responses.
+
+#### Sample Rest.li server 
+The ```restli-testsuite-server``` directory contains code for a Rest.li Java server. Using the Java resources and keys,
+Java Rest.li will generate language-independent restspecs in the ```client-testsuite/restspecs``` folder. Using these restspecs,
+other Rest.li implementations can generate request builders without modifying server code.
+When adding a new test, you may want to update or add a resource to ```restli-testsuite-server/src/main/java/testsuite```.
+See the section on wire protocol tests in [How To Run the Java TestNG Tests and Expand the Test Suite Specification](testsuite_how_to.md).
+
+Rest.li Test Suite Specification Coverage
+------------------
+The test suite spec is intended to cover three categories of Rest.li client behavior: **JSON serialization, data template
+generation, and wire protocol**. Each test category is described in more detail below. 
+  
+### JSON Tests
+These are tests for serialization and deserialization. Tests cover basic JSON and JSON corner cases such as large
+numbers, special characters and encodings.
+
+
+| JSON feature | Details |
+|--------------|---------|
+| Basic Types| string, number, boolean, null, object, array|
+| Empty Collections| empty object, empty array, object with empty arrays, object with empty objects, array of empty object, array of empty arrays| 
+| Large Numbers |i32 and i64|
+| Special Characters| periods in key|
+| Unicode| Chinese character and e with an accent mark|
+
+
+### Data Schema Tests
+These are tests for data template generation from schema. Tests cover schema types (records, unions, enums, typerefs,
+...), primitive types, optionals and defaults. Backward compatibility rules are also covered.
+                                                            
+
+| Schema feature | Details|
+|------------------|--------|
+| Primitive Types|int, long, float, double, bytes, string |
+| Complex Types | array of maps, map of ints, record with props |
```

Comment:
record with props? What props mean here? just fields in record? Then we can just say "record"